src/loader.js

```diff
@@ -3,6 +3,53 @@ import path from 'path';
 import gParser from 'gettext-parser';
 import { addLocale } from './index';
 
+
+const REG = /\$\{ \w+(.\w+)* \}/g;
+
+function replaceVariables(str, obj) {
+    return str.replace(REG, (variable) => {
+        return `\$\{ ${obj[variable]} \}`;
+    });
+}
+
+function transformTranslate(translate) {
+    const variables = translate.msgid.match(REG);
+    if (!variables) {
+        return translate;
+    }
+
+    const variableNumberMap = {};
+    variables.forEach((variable, i) => {
+        variableNumberMap[variable] = i;
+    });
+
+    const msgid = replaceVariables(translate.msgid, variableNumberMap);
+
+    const newTranslate = { msgid };
+
+    if (translate.msgid_plural) {
+        newTranslate.msgid_plural = replaceVariables(translate.msgid_plural, variableNumberMap);
+    }
+
+    newTranslate.msgstr = [];
+    for (const str of translate.msgstr) {
+        newTranslate.msgstr.push(replaceVariables(str, variableNumberMap));
+    }
+    newTranslate.comments = translate.comments;
+    return newTranslate;
+}
+
+function transformTranslateObj(translateObj) {
+    const newTranslations = {};
+    const translations = translateObj.translations[''];
+    for (const msgid of Object.keys(translations)) {
+        const newTranslate = transformTranslate(translations[msgid]);
+        newTranslations[newTranslate.msgid] = newTranslate;
+    }
+    translateObj.translations[''] = newTranslations;
```

Comment:
There is a plan to support contexts. So `translations['']` with empty string is just a default context. We should iterate through `translateObj.translations` keys and reassign them with new translations, but not hard code them.